### PR TITLE
add BLOM regression test functionality

### PR DESCRIPTION
**Pull request by mvertens**
This PR has changes to buildnml and to the testdefs directory to enable BLOM to have its first set of regression tests.
This PR resolves issue #258.

**To see how the regression test is working - do the following:** 
(Note: the BLOM external points to my repository until this PR is merged)

```
> git clone https://github.com/mvertens/NorESM.git
> cd NorESM
> git checkout noresm_develop_v7_proto
> cd cime/scripts
> ./query_testlists --xml-category aux_blom_noresm
```
will give the following output

```
aux_blom_noresm: SMS_D_Ld1.T62_tn14.NOINYOC.betzy_intel
aux_blom_noresm: ERS_Ld5.T62_tn14.NOINYOC.betzy_intel
aux_blom_noresm: ERS_Ld5.T62_tn14.NOINY.betzy_intel
aux_blom_noresm: ERS_Ld5.T62_tn14_wtn14nw.NOINY_WW3.betzy_intel
aux_blom_noresm: ERS_Ld5.T62_tn14_wtn14nw.NOINY_WW3.betzy_intel.blom-wavice
```

**To run a test suite and also generate baselines:**

```
./create_test 
    --xml-category aux_blom_noresm  
    --xml-machine betzy --xml-compiler intel 
    --generate <your_baseline_directory_name>
    --baseline-root /cluster/shared/noresm/noresm_baselines 
    --project <your_project>
```
Baselines for the above aux_blom_noresm tests will be created in 
`      /cluster/shared/noresm/noresm_baselines/<your_baseline_directory_name>`

More documentation on regression testing in NorESM can be found at:
https://noresm-docs.readthedocs.io/en/noresm2/testing/systemtests.html

### CODE REVIEW

**Review by TomasTorsvik — APPROVED** [review]
I should look more into the regression test functionality, but I don't see any reason to hold back on merging this PR.

**Comment by mvertens** [comment]
@TomasTorsvik - thanks for the review. Do we need to wait for the review for Mats as well. I'd like to get this into a updated NORESM development snapshot that is needed for the Making Waves project. This is the last branch that is still pointing to my own forked code.

**Comment by TomasTorsvik** [comment]
> @TomasTorsvik - thanks for the review. Do we need to wait for the review for Mats as well. I'd like to get this into a updated NORESM development snapshot that is needed for the Making Waves project. This is the last branch that is still pointing to my own forked code.

@mvertens - I usually wait for review from Mats when a PR touches physics or model coupling (i.e. not purely BGC). Since this PR is mainly adding test functionality it is different from the PRs that we usually deal with. If you need this functionality in your current work, I suggest we merge it so that it doesn't stop you in your progress. Otherwise I would prefer to wait for input from Mats.

**Pull request merged**
